Include/Rocket/Core/WString.h

```diff
@@ -47,6 +47,9 @@ class ROCKETCORE_API WString : public WStringBase
 public:
 	/// Constructs an empty string.
 	WString();
+	WString( WString const& ) = default;
+	WString( WString && ) = default;
+	WString& operator=( WString && ) = default;
```

Comment:
`StringBase` has raw pointer members, and does not define a move constructor. So this seems unlikely to do the right thing.

In general I don't suggest trying to hack on librocket since the work will be wasted when it is replaced.

Reply:
Well I should say, it will probably use the copy constructor of StringBase. So it won't exactly do a *wrong* thing, but it certainly won't do a *useful* thing.

Reply:
What this does is silence a warning which is emitted a lot. It does not alter the code's behavior, since what was the standard before C++11 was that if a constructor was missing, then the default generated one would be used.

So, while I agree that's it's not useful per-se, it still helps by reducing noise at build time, making it easier to notice useful warnings.

Reply:
Just get rid of the move assignment operator and I'd be fine with it

Reply:
Then I should make it a `= delete` I guess? Because just removing it won't always fix the warning, and will still make the same code being run in practice.

Reply:
You may optionally define deleted move construction and move assignment operators. It doesn't actually have an effect because if you explicitly define a copy constructor or assignment operator, then the move constructor and move assignment operator are deleted by default.

Reply:
I am not aware of such rule, but I'm not a C++ lawyer. In any case, g++ fails to build if I mark them deleted. Simply removing the definition do silence the warning, though, so let's do that.

Reply:
LGTM